packages/waku/src/router/create-pages.ts

```diff
@@ -568,6 +606,41 @@ export const createPages = <
         ),
       };
     },
+    getApiConfig: async () => {
+      await configure();
+
+      return Array.from(apiPathMap.values()).map(({ pathSpec, mode }) => {
+        return {
+          path: pathSpec,
+          isStatic: mode === 'static',
+        };
+      });
+    },
+    handleApi: async (path, request) => {
+      await configure();
+      const routePath = getRoutePath(path);
+      if (!routePath) {
+        throw new Error('Route not found: ' + path);
+      }
+      const { handler } = apiPathMap.get(routePath)!;
+
+      const req = new Request(
+        new URL(
+          path,
+          typeof window !== 'undefined'
+            ? window.location.href
+            : 'http://localhost', // FIXME what should this be?
```

Comment:
@dai-shi do you know how this should be handled? `Request` needs a valid url, which seems to require a host. When we run the tests `localhost` is the "host".

Reply:
I think having always `localhost` is okay. When we learn more use cases, we can reconsider. As a workaround, we could use `unstable_getHeaders` for "host".

```js
new URL(path, 'http://localhost')
```

Reply:
hm I get this when calling `unstable_getHeaders` from `createPages.handleApi`

<img width="1004" alt="image" src="https://github.com/user-attachments/assets/f762f699-f324-48eb-81de-afd1bf9459d5" />

I'm ok with leaving it as localhost though.

Reply:
Hmm, I'm not sure why it errors. let's look into it separately.

Actually, you can probably read "host" header from params.headers, but if it's not available, we can fallback to `localhost`.

Reply:
like this?

```ts
      const req = new Request(
        new URL(path, options.headers['host'] ?? 'http://localhost'),
        options,
      );
```

Reply:
I'm not sure but should we need the protocol part too?

Reply:
<img width="557" alt="image" src="https://github.com/user-attachments/assets/0b1f42f5-9593-4631-a0cb-4e1e9c554cfc" />

I think it's needed? This is the output when run through my browser console

Reply:
Yes, I mean the same and `options.headers['host']` doesn't include the protocol part, right?

Reply:
Good point about the protocol. 

However, handApi will run on our server, right?

So will we always want to request to http://localhost actually? I think it makes sense to remove the use of the host header since we expect this to request from the server this code executes on.

Correct me if I'm off here though.

Reply:
It runs on our server (dynamic) or the build time (static).

I could imagine some use cases would check "host" and change the behavior based on it. So, at least they should be able to read the "host" header.
At that point, creating `URL` with the "host" header makes sense, but we can defer the decision until we see clear requirements.